src/nvim/eval/userfunc.c

```diff
@@ -3417,7 +3416,7 @@ hashitem_T *find_hi_in_scoped_ht(const char *name, hashtab_T **pht)
   while (current_funccal != NULL) {
     hashtab_T *ht = find_var_ht(name, namelen, &varname);
     if (ht != NULL && *varname != NUL) {
-      hi = hash_find_len(ht, varname, namelen - (varname - name));
+      hi = hash_find_len(ht, varname, (size_t)((ptrdiff_t)namelen - (varname - name)));
```

Comment:
`(varname - name)` is non negative so you could just `(size_t)` directly there

Reply:
Also a good catch :)